zebra/rt_netlink.c

```diff
@@ -2021,6 +2021,23 @@ static int netlink_neigh_update(int cmd, int ifindex, void *addr, char *lla,
 	if (lla)
 		nl_attr_put(&req.n, sizeof(req), NDA_LLADDR, lla, llalen);
 
+	if (IS_ZEBRA_DEBUG_KERNEL) {
+		char ip_str[INET6_ADDRSTRLEN + 8];
+		struct interface *ifp = if_lookup_by_index_per_ns(
+			zebra_ns_lookup(ns_id), ifindex);
+		if (family == AF_INET6)
+			snprintfrr(ip_str, sizeof(ip_str), "ipv6 %pI6",
+				   (struct in6_addr *)addr);
+		else
+			snprintfrr(ip_str, sizeof(ip_str), "ipv4 %pI4",
+				   (in_addr_t *)addr);
+		zlog_debug(
+			"%s: %s ifname %s ifindex %u addr %s mac %pEA vrf %s(%u)",
+			__func__, nl_msg_type_to_str(cmd),
+			ifp ? ifp->name : "null", ifindex, ip_str,
+			(struct ethaddr *)lla, ifp ? ifp->vrf->name : "null",
+			ifp ? ifp->vrf->vrf_id : 0);
```

Comment:
We should reuse `vrf_id_to_name()`.

Reply:
used vrf_id_to_name() to print. Please find the  tested output

2023/02/27 09:10:39 ZEBRA: [RT79T-F8QAW] netlink_neigh_update: RTM_DELNEIGH ifname vet
h0 **ifindex 5 addr ipv4 169.254.0.1 mac 86:3d:f2:9a:72:e6 vrf default(0)**
2023/02/27 09:10:39 ZEBRA: [HCFTH-1NSTK] netlink_talk: netlink-cmd (NS 0) type RTM_DEL
NEIGH(29), len=56 seq=12 flags 0x401
2023/02/27 09:10:39 ZEBRA: [MQ5AP-2S1F5] netlink-cmd (NS 0) error: No such file or dir
ectory, type=RTM_DELNEIGH(29), seq=12, pid=3535716854
2023/02/27 09:10:39 ZEBRA: [RT79T-F8QAW] netlink_neigh_update: RTM_NEWNEIGH ifname vet
h0 ifindex 5 addr ipv4 169.254.0.1 mac 86:3d:f2:9a:72:e6 vrf default(0)
2023/02/27 09:10:39 ZEBRA: [HCFTH-1NSTK] netlink_talk: netlink-cmd (NS 0) type RTM_NEW
NEIGH(28), len=56 seq=13 flags 0x401